Neos.ContentRepository/Classes/Security/Authorization/Privilege/Node/NodePrivilegeContext.php

```diff
@@ -159,6 +176,42 @@ public function isInWorkspace($workspaceNames)
         return in_array($this->node->getWorkspace()->getName(), $workspaceNames);
     }
 
+    /**
+     * Matches if the currently-selected workspace is the target workspace of the current user.
+     *
+     * Example: userIsInTargetWorkspace(['preview-1234']) matches if the current user
```

Comment:
I'm not 100% sure about the naming tbh.
*is* the user really in the target workspace?

What about just `isTargetWorkspace()`?

Reply:
The also not 100 % sure. But `isTargetWorkspace()` is not what this matcher will test IMO, as it is (in theory) a matcher on the node and not on the user. 

We could use sth like `canPublishToTargetWorkspace()` or sth. similar if we want to remove the `user`. I'm in for new suggestions though.